File: app/src/main/java/ca/ivasenko/mobiletest/categories/ui/MainActivity.java
package ca.ivasenko.mobiletest.categories.ui;

import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import java.util.List;

import ca.ivasenko.mobiletest.R;
import ca.ivasenko.mobiletest.categories.model.Category;

public class MainActivity extends AppCompatActivity
{
    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        RecyclerView recyclerView = findViewById(R.id.recyclerview);
        int tilePadding = getResources().getDimensionPixelSize(R.dimen.tile_padding);
        recyclerView.setPadding(tilePadding, tilePadding, tilePadding, tilePadding);
        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);
        DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(recyclerView.getContext(),
                layoutManager.getOrientation());
        recyclerView.addItemDecoration(dividerItemDecoration);

        final CategoriesContentAdapter adapter = new CategoriesContentAdapter(this);
        recyclerView.setAdapter(adapter);

        CategoryListViewModel categoryListViewModel = ViewModelProviders.of(this).get(CategoryListViewModel.class);
        categoryListViewModel.getAllCategories().observe(this, new Observer<List<Category>>()
        {
            @Override
            public void onChanged(@Nullable List<Category> categories)
            {
                adapter.setCategories(categories);
            }
        });
    }
}

File: CodeTest/app/src/main/java/com/quickseries/codetest/MainActivity.kt
package com.quickseries.codetest

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import com.quickseries.codetest.adapter.RecyclerCategoriesAdapter
import com.quickseries.codetest.model.Categorie
import com.quickseries.codetest.services.CategoriesService.getCategories
import com.quickseries.codetest.util.jsonUtil.loadJSONFromAsset
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    private var categories: MutableList<Categorie> = mutableListOf()
    lateinit var categorieSorted: List<Categorie>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        title = "Categories"


        val json = loadJSONFromAsset(application, "categories.json")
        categories = getCategories(json!!)
        categorieSorted = categories.sortedWith(compareBy {it.title})
        
        setupCategoriesRecycler()
    }

    private fun setupCategoriesRecycler() {
        categories_recycler.adapter = RecyclerCategoriesAdapter(categorieSorted)
        categories_recycler.layoutManager = LinearLayoutManager(this@MainActivity, LinearLayoutManager.VERTICAL, false)
        categories_recycler.addItemDecoration(DividerItemDecoration(this, DividerItemDecoration.VERTICAL))
    }

}
